File: scripts/yaml2json.py
#!/usr/bin/env python3

# SPDX-FileCopyrightText: None
#
# SPDX-License-Identifier: CC0-1.0

import json
from pathlib import Path
from typing import Final

import yaml

ASSETS_DIR: Final[Path] = Path(__file__).resolve().parent.parent / "assets"
YAML_DIR: Final[Path] = ASSETS_DIR / "yaml"
JSON_DIR: Final[Path] = ASSETS_DIR / "json"

for yaml_file in YAML_DIR.iterdir():
    with yaml_file.open("rb") as f:
        data = yaml.safe_load(f)

    json_file = JSON_DIR / (yaml_file.stem + ".json")
    with json_file.open("w") as f:
        json.dump(data, f, ensure_ascii=False, indent=2, default=str)

File: scripts/yaml2json.ts
#!/usr/bin/env -S deno run --allow-read --allow-write

// SPDX-FileCopyrightText: 2024 Shun Sakai
//
// SPDX-License-Identifier: CC0-1.0

import * as path from "@std/path";
import * as yaml from "@std/yaml";

const assetsDir = path.join(
  path.dirname(path.dirname(path.fromFileUrl(import.meta.url))),
  "assets",
);
const yamlDir = path.join(assetsDir, "yaml");
const jsonDir = path.join(assetsDir, "json");

for await (const dirEntry of Deno.readDir(yamlDir)) {
  const yamlFile = path.join(yamlDir, dirEntry.name);
  const inputText = await Deno.readTextFile(yamlFile);
  const data = yaml.parse(inputText);

  const jsonFile = path.join(jsonDir, path.parse(yamlFile).name + ".json");
  const jsonString = JSON.stringify(data, null, 2);
  await Deno.writeTextFile(jsonFile, jsonString);
}
